### PR TITLE
Fix pcolormesh warnings

### LINKED ISSUE #115: Plot shading flat is deprecated
**Describe the bug**

**To Reproduce**
1. gt.plot(x, y, grid)
Where grid is a 2D array with dimensions x and y

**Expected behavior**
Not to cause a warning. Note that this warning will graduate to an error soon!

**Screenshots**
image

I'm pretty sure this could be fixed with the addition of `shading='auto'` as a default kwarg to pcolormesh:

image

Not 100 % sure on implementation of this though, as it's in the `plot_functions` class

### PR DESCRIPTION
**Pull request by dhruvbalwada**
- [ ] Closes #115
 - [ ] Tests added: test_plot
 - [ ] Passes `pre-commit run --all-files`

I have added the rasterized and shading as default kwargs, so no more warnings should be raised. warnings are not being raised at the moment. 

I have also added some tests that should be unhappy if warnings are raised. I think it works, because it raised a `failed` if I tried without my default kwargs. Please double check.

### CODE REVIEW

File: tests/test_plot.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# load some data
+filenames = "./tests/data/p542*.nc"
+
+names = ["ctd_depth", "ctd_time", "ctd_pressure", "salinity", "temperature"]
+ds_dict = seaglider_basestation_netCDFs(filenames, names, keep_global_attrs=False)
+
+dat = ds_dict["sg_data_point"]
+
+# start tests
+def test_dummy():
```
Comment (jbusecke) [inline tests/test_plot.py:17]: Can you remove this one? I just put it there as a placeholder until we got real tests.

File: tests/test_plot.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    """WE REALLY NEED TO ADD TESTS!!! THESE JUST TEST THE BASIC IMPORT!!!"""
+    assert 1 == 1
+
+
+def test_no_warms():
```
Comment (jbusecke) [inline tests/test_plot.py:22]: You can easily check if this works by deactivating the added shading option (e.g. commenting it out temporarily) and then run the test. It should fail in that instance.

**Review by jbusecke — CHANGES_REQUESTED** [review]

**Comment by jbusecke** [comment]
> I have also added some tests that should be unhappy if warnings are raised. I think it works, because it raised a failed if I tried without my default kwargs. Please double check.

Disregard my comment in the review above. You already did it! That is sufficient in my eyes.

So as soon as the dummy test is removed, we can get this merged. Thanks @dhruvbalwada

**Comment by jbusecke** [comment]
I also renamed `warms` to `warns` in the test name. Hope that is ok.

**Comment by dhruvbalwada** [comment]
Thanks for taking a look and the fixes. Yes warms sounds fluffy and nice, but warns is probably more appropriate.

**Pull request merged**